content/blog/2019/egress-traffic-control-in-istio-part-2/index.md

```diff
@@ -0,0 +1,116 @@
+
+After all, if the attackers are able to attack Istio Mixer or the egress gateway, they could
+violate any policy.
+
+Once you direct egress traffic through an egress gateway and apply the additional security mechanisms,
+you can securely monitor and enforce security policies for the traffic.
+
+The following diagram shows Istio's security architecture, augmented with an L3 firewall which is part of the
+[additional security mechanisms](/docs/tasks/traffic-management/egress/egress-gateway/#additional-security-considerations)
+that should be provided outside of Istio.
+
+{{< image width="80%" link="./SecurityArchitectureWithL3Firewalls.svg" caption="Istio Security Architecture with Egress Gateway and L3 Firewall" >}}
+
+You can configure the L3 firewall trivially to only allow incoming traffic through the Istio ingress gateway and
+only allow outgoing traffic through the Istio egress gateway. The Istio proxies of the gateways enforce policies and report telemetry as all other proxies in the mesh.
+
+Now let's examine possible attacks and let me show you how the secure control of egress traffic in Istio prevents them.
+
+## Preventing possible attacks
+
+Consider the following security policies for egress traffic:
+
+* Application **A** is allowed to access `*.ibm.com`, which includes all the external services with URLs matching
+`*.ibm.com`.
+* Application **B** is allowed to access `mongo1.composedb.com`.
+* All egress traffic is monitored.
+
+Suppose the attackers have the following goals:
+
+1. To access `*.ibm.com` from your cluster.
+1. To access `*.ibm.com` from your cluster, unmonitored. The attackers want their traffic to be unmonitored to prevent a
+possibility that you will detect the forbidden access.
+1. To access `mongo1.composedb.com` from your cluster.
+
+Now suppose that the attackers manage to break into one of the pods of application **A**. The attackers try to use the compromised pod to perform the forbidden access.
+
+First, let's see which attacker's goals it is possible for you to thwart:
+
+1. Since application **A** is allowed to access `*.ibm.com`, the compromised pod is able to access it. There is no way to prevent such access since there is no way to distinguish, at
+least initially, between the original and the compromised versions of the pod.
+1. Fortunately, if you can monitor all
+access to external services, you could detect suspicious traffic and thwart the second goal of the attackers.
+For example, you could apply anomaly detection tools on the egress traffic logs.
+1. You can thwart the third goal if you can correctly detect the source of the traffic, in this case, application **A**,
+and check that it is not allowed to access the destination, `mongo1.composedb.com`, according to the security policies
+mentioned above.
+
+Now, let's see which attacks malicious actors could attempt to achieve their goals and how secure control of egress
```

Comment:
It's confusing to readers that you now move to other attack vectors. How do these relate to the attacks you mentioned on the previous list? It becomes even more confusing when you think about how the previous text mentioned "compromised pods" and "unmonitored access" which would fall under the categories you mention here, "Compromise" and "Bypass" respectively. Please merge the two lists into a single three Item list: Bypass, Compromise, Impersonate. Each item on the list should include: The description of the attack using the examples and the features and/or configurations that prevent the attack.

Reply:
@rcaballeromx Please note that there the attackers have three goals and they use three different attacks to achieve the goals. The number three is accidental, the number of possible attacks does not have to correspond to the number of goals.

Reply:
I rewrote that part in https://github.com/istio/istio.io/pull/4232/commits/1a82f2e5afeeb9d3a49212f7c3ff34fb47cf6bbd. The topic is tricky. The first list is related to the attackers trying to perform the forbidden access in a straightforward way, without any tricks. The second list is related to the tricks the attackers can perform.

Analogy to the real world:

Consider an airport policy: anyone from a wanted list of the police is not allowed to enter the airport building.
The attacker is in the wanted list of the police. His goal is to enter the airport building.
Security measures: put a police officer at the entrance to the building. Lock all the employees entrances.
The attacker can try his luck and to achieve his goal in a straightforward way - to enter through the main entrance. There he will be stopped by the police officer. The attacker can try a trick - to enter the building through an employees entrance. But since it is locked, this attack is prevented.